src/solutions/ode_solutions.jl

```diff
@@ -613,7 +613,9 @@ function strip_solution(sol::ODESolution)
 
     interp = strip_interpolation(sol.interp)
 
-    ODESolution(sol.u, sol.u_analytic, sol.errors,
+    bigT = typeof(sol).parameters[1]
+    bigN = typeof(sol).parameters[2]
```

Comment:
This is not a good idea, it will not infer.

Reply:
Instead use `@set sol.interp = interp`

Reply:
Where does `@set` come from? When I do 
```julia
 @set sol.interp = interp
   
    ODESolution(sol.u, sol.u_analytic, sol.errors,
        sol.t, sol.k, sol.discretes, nothing, nothing,
        sol.interp, sol.dense, sol.tslocation, sol.stats,
        sol.alg_choice, sol.retcode, sol.resid, sol.original)
```
It still gives a method error.

Reply:
Would something like this be better?
```julia
N = sol.u === nothing ? 2 : ndims(eltype(sol.u)) + 1
T = eltype(eltype(sol.u))
```

Reply:
Oh I see, from Accessors